fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/mappers/DateTimeSortMapper.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.metric.mappers;
+
+import com.yahoo.bard.webservice.data.Result;
+import com.yahoo.bard.webservice.data.ResultSet;
+import com.yahoo.bard.webservice.druid.model.orderby.SortDirection;
+import com.yahoo.bard.webservice.table.Schema;
+
+import org.joda.time.DateTime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Default dateTime order of the druid results is ascending. This class reverses the order if user requests.
+ */
+public class DateTimeSortMapper extends ResultSetMapper {
+
+    private SortDirection direction;
+
+    /**
+     * Constructor.
+     *
+     * @param direction Sort direction
+     */
+    public DateTimeSortMapper(SortDirection direction) {
+        this.direction = direction;
+    }
+
+    /**
+     *  Reverses the resultSet if the requested order is DESC.
+     *
+     * @param resultSet  The result set need to be sorted in descending order
+     *
+     * @return sorted ResultSet
+     */
+    @Override
+    public ResultSet map(ResultSet resultSet) {
+
+        Map<DateTime, List<Result>> bucketizedResultsMap = new LinkedHashMap<>();
+
+        for (Result result: resultSet) {
+            bucketizedResultsMap.computeIfAbsent(result.getTimeStamp(), ignored -> new ArrayList<>()).add(result);
+        }
+
+        List<DateTime> dateTimeList = new ArrayList(bucketizedResultsMap.keySet());
+
+        Collections.sort(dateTimeList, new DateTimeComparator());
+
+        if (direction.equals(SortDirection.DESC)) {
+            Collections.sort(dateTimeList, Collections.reverseOrder());
+        }
+
```

Comment:
This doesn't make much sense.  You're doing a comparison based sort.  Then doing another one.